Patch by Sergei Golovan fixes race condition during ejabberd start when
error messages from ejabberd_listener didn't reach a logger because it
hadn't started yet.

Index: ejabberd-1.1.4/src/ejabberd_app.erl
===================================================================
--- ejabberd-1.1.4.orig/src/ejabberd_app.erl
+++ ejabberd-1.1.4/src/ejabberd_app.erl
@@ -22,6 +22,7 @@
     db_init(),
     sha:start(),
     catch ssl:start(),
+    start(),
     stringprep_sup:start_link(),    
     translate:start(),
     acl:start(),
@@ -35,7 +36,6 @@
     %eprof:start(),
     %eprof:profile([self()]),
     %fprof:trace(start, "/tmp/fprof"),
-    start(),
     load_modules(),
     Sup;
 start(_, _) ->
@@ -45,10 +45,6 @@
     ok.
 
 start() ->
-    spawn_link(?MODULE, init, []).
-
-init() ->
-    register(ejabberd, self()),
     %erlang:system_flag(fullsweep_after, 0),
     %error_logger:logfile({open, ?LOG_PATH}),
     LogPath =
@@ -64,6 +60,10 @@
 		end
 	end,
     error_logger:add_report_handler(ejabberd_logger_h, LogPath),
+    spawn_link(?MODULE, init, []).
+
+init() ->
+    register(ejabberd, self()),
     erl_ddll:load_driver(ejabberd:get_so_path(), tls_drv),
     case erl_ddll:load_driver(ejabberd:get_so_path(), expat_erl) of
 	ok -> ok;
